tsconfig.json

```diff
@@ -0,0 +1,24 @@
+{
```

Comment:
as this is the first time we use typescript and this config may define how we do it in other projects in the future: can you provide some more context on why you chose to configure it like this?

especially for cases where you deviated from whatever the 'standard' is (if there is one)

Reply:
TBH most of it is copied together from 'Getting started with TS' articles. I'd prefer to do this fine-tuning in a different PR. 

These two seem required for libraries 
```
"declaration": true,
"declarationMap": true,
```

If you're already aware of better settings here, happy to discuss/update already. Otherwise, I'd update this as we go.

Reply:
i'm not, was just wondering whether there were any conscious decisions you took here that deviate from a standard because of some tricks you've learned about TypeScript :)